### PR TITLE
fix (monitoring): fix SQL query

### PR DESCRIPTION
**Pull request by callapa**
## Description

Some queries are 

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 22.10.x
- [x] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/class/centreonMonitoring.class.php [hunk 1]
```diff
@@ -76,61 +85,92 @@ public function getPoller()
...
         }
 
-        if (!$objXMLBG->is_admin) {
-            $rq .=  "AND h.host_id = centreon_acl.host_id "
-                . "AND s.service_id = centreon_acl.service_id "
-                . "AND centreon_acl.group_id IN (" .  $obj->access->getAccessGroupsString() . ") ";
+        # Acknowledgement filter
+        if ($o === 'svcSum_ack_0') {
+            $query .= ' AND s.acknowledged = 0 AND s.state != 0 ';
+        } elseif ($o === "svcSum_ack_1") {
```
Comment (Drago37) [inline centreon/www/class/centreonMonitoring.class.php:140]: You can use constants no ?

File: centreon/www/class/centreonMonitoring.class.php [hunk 2]
```diff
@@ -76,61 +85,92 @@ public function getPoller()
...
+        $statement = $centreonXMLBGRequest->DBC->prepare($query);
+        $centreonXMLBGRequest->DBC->executePreparedQuery($statement, $toBind);
 
-            $DBRESULT = $objXMLBG->DBC->query($rq);
-
-            $cpt = 0;
-        if ($DBRESULT->rowCount()) {
-            $row = $DBRESULT->fetchRow();
-            $cpt = $row['count'];
+        if (($count = $statement->fetchColumn()) !== false) {
```
Comment (Drago37) [inline centreon/www/class/centreonMonitoring.class.php:146]: Now, it's recommended to use fetchColumn of CentreonDB to have a better error handling.

File: centreon/www/class/centreonMonitoring.class.php [hunk 3]
```diff
@@ -145,53 +185,57 @@ public function getServiceStatus($hostList, $objXMLBG, $o, $instance, $hostgroup
...
+        $serviceDetails = [];
+        $statement = $centreonXMLBGRequest->DBC->prepare($query);
+        $centreonXMLBGRequest->DBC->executePreparedQuery($statement, $toBind);
 
-        $tab = [];
-        $DBRESULT = $objXMLBG->DBC->query($rq);
-        while ($svc = $DBRESULT->fetchRow()) {
-            if (!isset($tab[$svc["name"]])) {
-                $tab[$svc["name"]] = [];
+        while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (Drago37) [inline centreon/www/class/centreonMonitoring.class.php:229]: Now, it's recommended to use fetch of CentreonDB to have a better error handling. PDO::FETCH_ASSOC is not necessary too.

File: centreon/www/class/centreonMonitoring.class.php [hunk 4]
```diff
@@ -145,53 +185,57 @@ public function getServiceStatus($hostList, $objXMLBG, $o, $instance, $hostgroup
...
             }
-            $tab[$svc["name"]][$svc["service_name"]] = [
-                'state' => $svc["state"],
-                'service_id' => $svc['service_id']
+            $serviceDetails[$result["name"]][$result["service_name"]] = [
+                'state' => $result["state"],
+                'service_id' => $result['service_id']
             ];
         }
-        $DBRESULT->closeCursor();
```
Comment (Drago37) [inline centreon/www/class/centreonMonitoring.class.php:193]: You can use closeQuery() of CentreonDB to close the query after the fetch() ;)

**Review by Drago37 — CHANGES_REQUESTED** [review]

File: centreon/www/class/centreonMonitoring.class.php (hunk 1 above)
Comment (callapa) [reply centreon/www/class/centreonMonitoring.class.php:140]: Yes, as long as you take the time to understand the function as a whole. Which I don't want to do at the moment.

File: centreon/www/class/centreonMonitoring.class.php (hunk 4 above)
Comment (callapa) [reply centreon/www/class/centreonMonitoring.class.php:193]: Yes but for this use case the closeQuery is useless.

File: centreon/www/class/centreonMonitoring.class.php (hunk 1 above)
Comment (Drago37) [reply centreon/www/class/centreonMonitoring.class.php:140]: I understand, it's better to have good names for the constants :D T resolve.

File: centreon/www/class/centreonMonitoring.class.php (hunk 2 above)
Comment (Drago37) [reply centreon/www/class/centreonMonitoring.class.php]: Thanks for the change :)

File: centreon/www/class/centreonMonitoring.class.php (hunk 3 above)
Comment (Drago37) [reply centreon/www/class/centreonMonitoring.class.php]: Thanks for the change :)

File: centreon/www/class/centreonMonitoring.class.php (hunk 4 above)
Comment (Drago37) [reply centreon/www/class/centreonMonitoring.class.php:193]: Why ? it seems to me that closeQuery also frees up memory, no ? I always close the query to be sure to not have errors and to make queries easier to read. But I know it's not obligatory too. So I resolve this thread :)

**Review by Drago37 — APPROVED** [review]
It's ok for me thanks for the changes :)

**Review by dmyios — APPROVED** [review]

**Pull request merged**